File: app/src/main/java/com/janyo/janyoshare/ViewPagerAdapter.java
package com.janyo.janyoshare;

import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import java.util.ArrayList;
import java.util.List;
import android.support.v4.app.Fragment;

public class ViewPagerAdapter extends FragmentPagerAdapter {

	private List<Fragment> fragmentList;
	private List<String> titleList;

	public ViewPagerAdapter(FragmentManager fm) {
		super(fm);
		fragmentList = new ArrayList<>();
		titleList = new ArrayList<>();
	}

	public void addFragment(Fragment fragment, String title) {
		fragmentList.add(fragment);
		titleList.add(title);
	}

	@Override
	public Fragment getItem(int position) {
		return fragmentList.get(position);
	}

	@Override
	public int getCount() {
		return fragmentList.size();
	}

	@Override
	public CharSequence getPageTitle(int position) {
		return titleList.get(position);
	}
}

File: app/src/main/java/com/janyo/janyoshare/ViewPagerAdapter.kt
package com.janyo.janyoshare

import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentPagerAdapter
import java.util.ArrayList
import android.support.v4.app.Fragment

class ViewPagerAdapter(fm: FragmentManager) : FragmentPagerAdapter(fm)
{

	private val fragmentList: MutableList<Fragment>
	private val titleList: MutableList<String>

	init
	{
		fragmentList = ArrayList<Fragment>()
		titleList = ArrayList<String>()
	}

	fun addFragment(fragment: Fragment, title: String)
	{
		fragmentList.add(fragment)
		titleList.add(title)
	}

	override fun getItem(position: Int): Fragment
	{
		return fragmentList[position]
	}

	override fun getCount(): Int
	{
		return fragmentList.size
	}

	override fun getPageTitle(position: Int): CharSequence
	{
		return titleList[position]
	}
}